### PR TITLE
Compatibility with ActiveRecord 5.0

### PR DESCRIPTION
**Pull request by bquorning**

### CODE REVIEW

**Comment by bquorning** [comment]
It looks like tables aren’t being created in Rails 5.0 tests. @grosser, @pschambacher & @osheroff would you mind taking a look?

**Comment by bquorning** [comment]
Actually, it looks like the way AR checks whether `table_exists?` has changed (rails/rails#21687, rails/rails#21601 etc.) That’s probably why the tests are failing.

**Comment by bquorning** [comment]
@grosser, @pschambacher, @osheroff ?

**Comment by grosser** [comment]
<- PTO not looking into crazy AR foobars ;)

**Comment by pschambacher** [comment]
OK I'm back from PTO and will look at this now ( @grosser it's nice to see that you're human after all :wink: )

**Comment by pschambacher** [comment]
@bquorning fixed
@grosser @osheroff review please?

**Comment by bquorning** [comment]
That last commit is weird. I wonder if the changes to AbstractMysqlAdapter between v4.2.5 and v5.0.0.beta1 is to blame.

**Comment by grosser** [comment]
:+1: looks innocent ...

**Comment by bquorning** [comment]
> looks innocent ...

It isn’t. `table_exists?` returns true for a table that only exists on the previously connected database. I fear it’ll lead to problems further down the road if we ignore this one.

**Comment by osheroff** [comment]
@bquorning can you write a test that exposes the problem more succinctly?   Rails is always dicking around with the location of its caches.

File: lib/active_record_host_pool/connection_adapter_mixin.rb [hunk 1]
```diff
@@ -57,6 +57,7 @@ def _switch_connection
         log("select_db #{_host_pool_current_database}", "SQL") do
           clear_cache! if respond_to?(:clear_cache!)
           raw_connection.select_db(_host_pool_current_database)
+          @config[:database] = _host_pool_current_database if ActiveRecord::VERSION::MAJOR >= 5
```
Comment (bquorning) [inline lib/active_record_host_pool/connection_adapter_mixin.rb]: Awesome! I still wonder if this should apply also to earlier versions of AR.

Comment (pschambacher) [inline lib/active_record_host_pool/connection_adapter_mixin.rb]: @grosser FYI `data_sources` and all methods based on it like `table_exists?` weren't working with the host pool gem.

**Comment by pschambacher** [comment]
:haircut:

**Comment by grosser** [comment]
:+1:

**Comment by pschambacher** [comment]
@osheroff @grosser can we have a release of the gem please? I bumped to v0.9.0 but can't release

**Pull request merged**